### PR TITLE
INGK-893 Recover EEPROM write functionality for manufacturing

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Add methods to read and write to the ESC EEPROM.

Fixes # INGK-893

### Type of change

- Add the read_eeprom method.
- Add the write_eeprom method.

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting and linting

- [x] Use the ruff package to format the code: `ruff format ingenialink tests`.
- [x] Use the ruff package to lint the code: `ruff check ingenialink`.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

**Comment by pzinemanas** [comment]
Please update CHANGELOG

File: ingenialink/ethercat/servo.py [hunk 1]
```diff
@@ -320,6 +323,49 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
             self.ETHERCAT_PDO_WATCHDOG, self.SECONDS_TO_MS_CONVERSION_FACTOR * timeout
         )
 
+    def read_eeprom(
+        self,
+        address: int,
+        length: int = DEFAULT_EEPROM_READ_BYTES_LENGTH,
```
Comment (pzinemanas) [inline ingenialink/ethercat/servo.py:329]: Should we check that length % 2 == 0?

File: ingenialink/ethercat/servo.py [hunk 2]
```diff
@@ -320,6 +323,49 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
...
+        return data
+
+    def write_eeprom(
+        self, address: int, data: bytes, timeout: int = DEFAULT_EEPROM_OPERATION_TIMEOUT_uS
+    ) -> None:
+        """Write to the ESC EEPROM.
+
+        Args:
+            address: EEPROM address to be written.
+            data: Data to be written. The data length must be a multiple of 2 bytes.
```
Comment (pzinemanas) [inline ingenialink/ethercat/servo.py:359]: Should we check that len(data) % 2 == 0?

File: ingenialink/ethercat/servo.py (hunk 1 above)
Comment (MartinALL91) [reply ingenialink/ethercat/servo.py:329]: It does not need to be a multiple of 2. Check the [test](https://github.com/ingeniamc/ingenialink-python/pull/447/files#diff-7e3660da5a421071fe4996b2da15a790b82f68bb1036f934ca9e8022e3692f66R9).

File: ingenialink/ethercat/servo.py (hunk 2 above)
Comment (MartinALL91) [reply ingenialink/ethercat/servo.py:359]: It is already checked by [PySOEM](https://github.com/bnjmnp/pysoem/blob/master/pysoem/pysoem.pyx#L821), but I could add it as well.

**Review by pzinemanas — APPROVED** [review]

**Comment by polfeliu** [comment]
Review the issue description. Manufacturing needs a file input just like twincat. You can request a sample to them

File: ingenialink/ethercat/servo.py [hunk 3]
```diff
@@ -320,6 +323,59 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
             self.ETHERCAT_PDO_WATCHDOG, self.SECONDS_TO_MS_CONVERSION_FACTOR * timeout
         )
 
+    def read_eeprom(
```
Comment (polfeliu) [inline ingenialink/ethercat/servo.py:326]: Users should not be using this and the name can lead to confusions.
The first thing a user will think of an eeprom of the drive is the eeprom of config.

I would call this read_esc_eeprom.

**Comment by MartinALL91** [comment]
> Review the issue description. Manufacturing needs a file input just like twincat. You can request a sample to them. Maybe it can be a different function that calls the write_eeprom that you wrote

Done

File: ingenialink/ethercat/servo.py [hunk 4]
```diff
@@ -320,6 +324,75 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
...
+        Returns:
+            EEPROM data. The read data.
+
+        Raises:
+            ValueError: If the length to be read has an invalid value.
+
+        """
+        if length < 1:
+            raise ValueError("The minimum length is 1 byte.")
+        data = bytes()
```
Comment (polfeliu) [inline ingenialink/ethercat/servo.py:349]: using a bytearray for appending will be a lot more efficient, but up to you

File: ingenialink/ethercat/servo.py [hunk 5]
```diff
@@ -320,6 +324,75 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
...
+        start_address = address
```
Comment (polfeliu) [inline ingenialink/ethercat/servo.py:350]: Is it necessary to create a new variable? In fact start_address increases on the loop so it's name loses sense

File: ingenialink/ethercat/servo.py [hunk 6]
```diff
@@ -320,6 +324,75 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
...
+        Raises:
+            ValueError: If the data has the wrong size.
+
+        """
+        if len(data) % 2 != 0:
+            raise ValueError("The data length must be a multiple of 2 bytes.")
+        start_address = address
+        while data:
+            self.slave.eeprom_write(start_address, data[:2], timeout)
+            data = data[2:]
```
Comment (polfeliu) [inline ingenialink/ethercat/servo.py:377]: There are ways to iterate over array 2 by 2 that are o(1) instead of this that is o(2)

Comment (MartinALL91) [reply ingenialink/ethercat/servo.py:377]: Essentially the same.
What do you suggest?

File: ingenialink/ethercat/servo.py (hunk 4 above)
Comment (MartinALL91) [reply ingenialink/ethercat/servo.py:349]: I could change it if you propose the necessary changes.

File: ingenialink/ethercat/servo.py (hunk 5 above)
Comment (MartinALL91) [reply ingenialink/ethercat/servo.py]: Done

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**